Data/test.lua

```diff
@@ -0,0 +1,9646 @@
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="2",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Versatile Performance",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_VersatilePerformance"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="2",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Well-Versed",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_WellVersed"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="5",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Lore Master",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_LoreMaster"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="10",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Jack of All Trades",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_JackOfAllTrades"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+  },
+  Name="Bard",
+  SpellStat="CHA",
+  Definitions={
+    {
+      Name="BardLVL",
+      InitialValue=0,
+    },
+    {
+      Name="CasterLevelBLBard",
+      InitialValue=0,
+    },
+  },
+  Bonuses={
+    {
+      Category="COMBAT",
+      Variables={
+        "BASEAB",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")*3/4",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateBABProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="SAVE",
+      Variables={
+        "BASE.Reflex",
+        "BASE.Will",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")/2+2",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateSaveProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="SAVE",
+      Variables={
+        "BASE.Fortitude",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")/3",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateSaveProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="VAR",
+      Variables={
+        "ClassBABModerate",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")",
```

Comment:
Weird.  This... doesn't seem right. Bards BAB is not equal to their class level? Is this an error in the source or an error in transcription?  I wonder if we should have a "dump a ton of debugging information in the output so I can track this back to where it came from" option for the generator... Or something like a PDB/JS source map?

Reply:
It looks like all of the Fractional stuff is wrong, and seems to just set everything to "classlevel".

Reply:
I think the variable it sets is what determines the stuff. Here it sets ClassBABModerate to "class level". The Fighter sets "ClassBABFull" to "class level". Its super strange.

Reply:
Oh weird, presumably those ClassBABModerate things are referenced... somewhere else to get your actual BAB. How oddly indirect...  What happens if you have 2 things that define the same "Varaible"? Are they added?

Reply:
I think they are processed in document order, so last one wins

Reply:
How... odd...  If you take 2 different levels of things that are BABModerate, say... cleric and rogue, they'd need to be summed together.  I'm really curious how all this fractional bonus stuff is implemented...

Reply:
Yea, its very complicated... for something that is super simple, `if (using fraction BAB) { round(sum(classes)) } else { sum(round(classes)) }`

Reply:
Yeah, seems like something the engine could just handle and doesn't need to be repeated in every class. Just indicating which of the 3 options the class is.

All of these variable must be scoped to the class entry itself... they would stomp all over each other otherwise... and something higher knows how to combine all the classes you have (add the BAB values, give you a spell list per class, etc)